src/main/java/pw/itr0/kaba/encrypt/PBE.java

```diff
@@ -12,65 +14,207 @@
         SecretKeyFactory keyFactory = SecretKeyFactory.getInstance(ENCRYPTION_ALGORITHM);
-        SecretKey key = keyFactory.generateSecret(keySpec);
+        SecretKey key;
+        try {
+            key = keyFactory.generateSecret(keySpec);
+        } catch (InvalidKeySpecException e) {
+            throw new MissImplementationException("Key specification must be type safely implemented. This exception must not occur.", e);
+        }
 
         PBEParameterSpec paramSpec = new PBEParameterSpec(salt, iterationCount);
 
-        Cipher cipher = Cipher.getInstance(ENCRYPTION_ALGORITHM);
-        cipher.init(mode, key, paramSpec);
+        Cipher cipher;
+        try {
+            cipher = Cipher.getInstance(ENCRYPTION_ALGORITHM);
+        } catch (NoSuchPaddingException e) {
+            throw new MissImplementationException("Padding algorithm must be statically implemented. This exception must not occur.", e);
+        }
+        try {
+            cipher.init(mode, key, paramSpec);
+        } catch (InvalidAlgorithmParameterException e) {
+            throw new MissImplementationException("Algorithm parameter must be type safely implemented. This exception must not occur.", e);
+        }
         return cipher;
     }
 
-    public static class Encrypter implements ByteEncrypter {
-
+    /**
+     * Encrypter for PBE.
+     *
+     * @author ryotan
+     * @since 1.0.0
+     */
+    public static class Encrypter implements PasswordBasedEncrypter {
+
+        /**
+         * Salt bytes of PBE encryption.
+         */
         private final byte[] salt;
+
+        /**
+         * iteration count of PBE encryption.
+         */
         private final int iterationCount;
 
+        /**
+         * Create PBE encrypter using given {@code salt} and {@code iterationCount}.
+         *
+         * @param salt           salt bytes of PBE encryption
+         * @param iterationCount iteration count of PBE encryption
+         */
         private Encrypter(byte[] salt, int iterationCount) {
-            this.salt = salt;
+            this.salt = new byte[salt.length];
+            System.arraycopy(salt, 0, this.salt, 0, salt.length);
             this.iterationCount = iterationCount;
         }
 
+        /**
+         * Encrypt secret bytes using password for PBE.
+         *
+         * @param password encrypting password
+         * @param secret   secret bytes
+         * @return PBE encrypted secret
+         * @throws GeneralSecurityException on encryption error
+         */
         @Override
         public byte[] encrypt(char[] password, byte[] secret) throws GeneralSecurityException {
             return getCipher(Cipher.ENCRYPT_MODE, password, salt, iterationCount).doFinal(secret);
         }
     }
 
-    public static class Decrypter implements ByteDecrypter {
+    /**
+     * Decrypter for PBE.
+     *
+     * @author ryotan
+     * @since 1.0.0
+     */
+    public static class Decrypter implements PasswordBasedDecrypter {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Class Decrypter should be declared as final.](https://www.codacy.com/app/ryotan/kaba-vault/file/2578430306/issues/source?bid=3298589&fileBranchId=3301306#l174)